network/network_windows.go

```diff
@@ -360,10 +360,20 @@ func (nm *networkManager) newNetworkImplHnsV2(nwInfo *EndpointInfo, extIf *exter
 		if errors.As(err, &hcn.NetworkNotFoundError{}) {
 			logger.Info("Creating hcn network", zap.Any("hcnNetwork", hcnNetwork))
 			hnsResponse, err = Hnsv2.CreateNetwork(hcnNetwork)
-			if err != nil {
-				return nil, fmt.Errorf("Failed to create hcn network: %s due to error: %v", hcnNetwork.Name, err)
+			if err == nil {
+				logger.Info("Successfully created hcn network with response", zap.Any("hnsResponse", hnsResponse))
+			} else {
+				if strings.Contains(err.Error(), "already exists") {
+					// fetch the network name again since the paralel CNI Add call has created the HNS network
+					hnsResponse, err = Hnsv2.GetNetworkByName(hcnNetwork.Name)
```

Comment:
no need to fetch again if cni gets already exists error.

Reply:
We need to fetch again to fill out hnsResponse since it is being used later on in the code

Reply:
ok